pandas/tests/reshape/test_concat.py

```diff
@@ -793,7 +800,32 @@ def test_append_records(self):
         expected = DataFrame(np.concatenate((arr1, arr2)))
         assert_frame_equal(result, expected)
 
-    def test_append_different_columns(self):
+    def test_append_sorts(self):
+        df1 = pd.DataFrame({"a": [1, 2], "b": [1, 2]}, columns=['b', 'a'])
+        df2 = pd.DataFrame({"a": [1, 2], 'c': [3, 4]}, index=[2, 3])
+        # default, changing in the future
+
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+            # from append we have an extra function call. Not worth hacking
+            # around to get the right stackleve.
+            result = df1.append(df2)
+
+        expected = pd.DataFrame({"b": [1, 2, None, None],
+                                 "a": [1, 2, 1, 2],
+                                 "c": [None, None, 3, 4]},
+                                columns=['a', 'b', 'c'])
+        tm.assert_frame_equal(result, expected)
+
+        # sort=True, the previous behavior
```

Comment:
you could just use the fixture and just change the expected if sort is False (like you do elsewhere)